File: be/src/main/java/plus/easydo/dnf/controller/DaMailController.java
package plus.easydo.dnf.controller;

import cn.dev33.satoken.annotation.SaCheckPermission;
import io.swagger.v3.oas.annotations.Operation;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import plus.easydo.dnf.dto.BatchSendResultDto;
import plus.easydo.dnf.dto.SendMailDto;
import plus.easydo.dnf.entity.DaMailSendLog;
import plus.easydo.dnf.entity.Postal;
import plus.easydo.dnf.qo.PageQo;
import plus.easydo.dnf.qo.RoleMailPageQo;
import plus.easydo.dnf.service.GameMailService;
import plus.easydo.dnf.service.IDaMailSendLogService;
import plus.easydo.dnf.vo.DataResult;
import plus.easydo.dnf.vo.R;

import java.util.List;

/**
 * 邮件发送记录 控制层。
 *
 * @author mybatis-flex-helper automatic generation
 * @since 1.0
 */
@RestController
@RequiredArgsConstructor
@RequestMapping("/api/mail")
public class DaMailController {

    private final IDaMailSendLogService daMailSendLogService;

    private final GameMailService gameMailService;


    @Operation(summary = "发送邮件")
    @SaCheckPermission("mail.sendMail")
    @PostMapping("/sendRoleMail")
    public R<Object> sendMail(@RequestBody SendMailDto sendMailDto) {
        gameMailService.sendMail(sendMailDto);
        return DataResult.ok();
    }

    @Operation(summary = "批量发送邮件")
    @SaCheckPermission("mail.sendMail")
    @PostMapping("/batchSendMail")
    public R<BatchSendResultDto> batchSendMail(@RequestBody SendMailDto sendMailDto) {
        BatchSendResultDto result = gameMailService.batchSendMail(sendMailDto);
        return DataResult.ok(result);
    }

    @Operation(summary = "异步批量发送邮件")
    @SaCheckPermission("mail.sendMail")
    @PostMapping("/asyncBatchSendMail")
    public R<String> asyncBatchSendMail(@RequestBody SendMailDto sendMailDto) {
        String taskId = gameMailService.asyncBatchSendMail(sendMailDto);
        return DataResult.ok(taskId);
    }

    @Operation(summary = "查询批量发送进度")
    @SaCheckPermission("mail.sendMail")
    @GetMapping("/batchSendProgress/{taskId}")
    public R<BatchSendResultDto> getBatchSendProgress(@PathVariable String taskId) {
        BatchSendResultDto result = gameMailService.getBatchSendProgress(taskId);
        return DataResult.ok(result);
    }

    @Operation(summary = "验证角色列表")
    @SaCheckPermission("mail.sendMail")
    @PostMapping("/validateCharacList")
    public R<List<Long>> validateCharacList(@RequestBody List<Long> characNoList) {
        List<Long> validList = gameMailService.validateCharacList(characNoList);
        return DataResult.ok(validList);
    }

    /**
     * 分页查询邮件发送记录
     *
     * @param pageQo 分页对象
     * @return 分页对象
     */
    @Operation(summary = "邮件发送记录分页")
    @SaCheckPermission("mail")
    @PostMapping("/pageMailSendLog")
    public R<List<DaMailSendLog>> pageMailSendLog(@RequestBody  PageQo pageQo) {
        return DataResult.ok(daMailSendLogService.sendLogpage(pageQo));
    }

    @Operation(summary = "角色邮件分页")
    @SaCheckPermission("mail")
    @PostMapping("/roleMailPage/{characNo}")
    public R<List<Postal>> roleMailPage(@PathVariable("characNo") Long characNo, @RequestBody RoleMailPageQo roleMailPageQo) {
        return DataResult.ok(gameMailService.roleMailPage(characNo,roleMailPageQo));
    }

    @Operation(summary = "删除邮件")
    @SaCheckPermission("mail")
    @PostMapping("/removeMail/{postalId}")
    public R<Boolean> removeMail(@PathVariable("postalId") Long postalId) {
        return DataResult.ok(gameMailService.removeMail(postalId));
    }

}

File: antd/src/services/dnf-admin/daMailController.ts
// @ts-ignore
/* eslint-disable */
import { request } from 'umi';

/** 邮件发送记录分页 POST /api/mail/pageMailSendLog */
export async function pageMailSendLog(body: API.PageQo, options?: { [key: string]: any }) {
  return request<API.RListDaMailSendLog>('/api/mail/pageMailSendLog', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 删除邮件 POST /api/mail/removeMail/${param0} */
export async function removeMail(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.removeMailParams,
  options?: { [key: string]: any },
) {
  const { postalId: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/mail/removeMail/${param0}`, {
    method: 'POST',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 角色邮件分页 POST /api/mail/roleMailPage/${param0} */
export async function roleMailPage(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.roleMailPageParams,
  body: API.RoleMailPageQo,
  options?: { [key: string]: any },
) {
  const { characNo: param0, ...queryParams } = params;
  return request<API.RListPostal>(`/api/mail/roleMailPage/${param0}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    params: { ...queryParams },
    data: body,
    ...(options || {}),
  });
}

/** 发送邮件 POST /api/mail/sendRoleMail */
export async function sendMail(body: API.SendMailDto, options?: { [key: string]: any }) {
  return request<API.RObject>('/api/mail/sendRoleMail', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 批量发送邮件 POST /api/mail/batchSendMail */
export async function batchSendMail(body: API.SendMailDto, options?: { [key: string]: any }) {
  return request<API.RBatchSendResultDto>('/api/mail/batchSendMail', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 异步批量发送邮件 POST /api/mail/asyncBatchSendMail */
export async function asyncBatchSendMail(body: API.SendMailDto, options?: { [key: string]: any }) {
  return request<API.RString>('/api/mail/asyncBatchSendMail', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 查询批量发送进度 GET /api/mail/batchSendProgress/${param0} */
export async function getBatchSendProgress(
  taskId: string,
  options?: { [key: string]: any },
) {
  return request<API.RBatchSendResultDto>(`/api/mail/batchSendProgress/${taskId}`, {
    method: 'GET',
    ...(options || {}),
  });
}

/** 验证角色列表 POST /api/mail/validateCharacList */
export async function validateCharacList(body: number[], options?: { [key: string]: any }) {
  return request<API.RListLong>('/api/mail/validateCharacList', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}
